core/net/rpl/rpl-ext-header.c

```diff
@@ -378,20 +388,20 @@ insert_srh_header()
     cmpri = MIN(cmpri, count_matching_bytes(&node_addr, &UIP_IP_BUF->destipaddr, 16));
     cmpre = cmpri;
 
-    PRINTF("RPL: SRH Hop ");
-    PRINT6ADDR(&node_addr);
-    PRINTF("\n");
+    if(node->parent != root_node) {
+      PRINTF("RPL: SRH Hop ");
+      PRINT6ADDR(&node_addr);
+      PRINTF("\n");
+    } else if(((DEBUG) & DEBUG_PRINT)) {
+      rpl_ns_get_node_global_addr(&node_addr, node);
+      PRINTF("RPL: SRH Next Hop ");
+      PRINT6ADDR(&node_addr);
+      PRINTF("\n");
+    }
     node = node->parent;
     path_len++;
   }
 
-  if(((DEBUG) & DEBUG_PRINT) && node != NULL) {
-    rpl_ns_get_node_global_addr(&node_addr, node);
-    PRINTF("RPL: SRH Next Hop ");
-    PRINT6ADDR(&node_addr);
-    PRINTF("\n");
-  }
```

Comment:
Due to the condition of `node != root_node` in `while` at line-373, `node` is identical to `root_node` at line-139. Then, you'll get not the address of the direct next hop but the root address in a debug message with "RPL: SRH Next Hop ".

Here is an example of the debug output. In this case, the node of ID:3 is the root and has "aaaa::203:3:3:3" as a global address.  You can see "aaaa::203:3:3:3" is printed as "SRH Next Hop" in the example.

> 00:00.073	ID:3	IPv6 addresses: aaaa::203:3:3:3
> 00:00.073	ID:3	fe80::203:3:3:3
> (snip)
> 00:11.349	ID:3	RPL: Packet going up, sender closer 0 (299 < 128)
> 00:11.349	ID:3	RPL: Rank OK
> 00:11.349	ID:3	RPL: SRH creating source routing header with destination aaaa::208:8:8:8 
> 00:11.349	ID:3	RPL: SRH Hop aaaa::207:7:7:7
> 00:11.349	ID:3	RPL: SRH Next Hop aaaa::203:3:3:3